### PR TITLE
chore: Begin introducing patterns for Multichain AssetList

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

This PR introduces some non-breaking patterns from https://github.com/MetaMask/metamask-extension/pull/28386

The idea is to introduce this as a smaller PR to reduce the diff in https://github.com/MetaMask/metamask-extension/pull/28386 to make it easier to review

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28429?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

App should be fully functional with no regressions

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jonybur — DISMISSED** [review]

File: ui/pages/asset/util.ts [hunk 1]
```diff
@@ -64,3 +64,31 @@ export const chainSupportsPricing = (chainId: `0x${string}`) =>
 
 /** The opacity components should set during transition */
 export const loadingOpacity = 0.2;
+
+// TODO: Add unit tests
```
Comment (darkwing) [inline ui/pages/asset/util.ts:68]: Once a few of these are in I'll approve!

Comment (gambinish) [reply ui/pages/asset/util.ts]: Added unit test here: https://github.com/MetaMask/metamask-extension/pull/28429/commits/1a1737a7a6f215464bc6c70c163a039e972a7a0a

File: ui/pages/asset/util.ts [hunk 2]
```diff
@@ -64,3 +64,31 @@ export const chainSupportsPricing = (chainId: `0x${string}`) =>
...
+
+  const tokens = data[chainId];
+
+  if (!tokens) {
+    console.warn(`No tokens found for chainId: ${chainId}`);
+    return null;
+  }
+
+  if (!address) {
+    return tokens.find((token) => !token.address);
```
[the remark this answers is not in the document]
Comment (gambinish) [reply ui/pages/asset/util.ts:86]: Hmm that's a good point. This case may not be correct. Let me address this

[the remark this answers is not in the document]
Comment (gambinish) [reply ui/pages/asset/util.ts:86]: There might be a case when the address is not provided, but we wouldn't want to grab the first matching token without an address in that case.

File: ui/pages/asset/util.ts (hunk 1 above)
Comment (gambinish) [reply ui/pages/asset/util.ts]: Updated with bugfix here: https://github.com/MetaMask/metamask-extension/pull/28429/commits/41adb561acf19278ba2a2f52fafa678d98f8e7d4

File: ui/pages/asset/util.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gambinish) [reply ui/pages/asset/util.ts]: Addressed: https://github.com/MetaMask/metamask-extension/pull/28429/commits/41adb561acf19278ba2a2f52fafa678d98f8e7d4

**Review by jonybur — DISMISSED** [review]

**Review by jonybur — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
Testing and all works great!

**Review by dabossdana — APPROVED** [review]

**Pull request merged**